spec/support/migration_spec_helper.rb

```diff
@@ -45,15 +45,8 @@ def migration_stub(klass)
   # Clears any cached column information on stubs, since the migrations
   # themselves will not expect anything to be cached.
   def clear_caches
-    ar_stubs.each do |s|
-      # inheritance_column changes done by migrations (e.g. to disable STI) are
-      # lost on reset_column_information, so we need to restore those changes.
-      i = s.inheritance_column
-
-      s.reset_column_information
-
-      s.inheritance_column = i
-    end
+    ar_stubs.each { |s| s.reset_column_information }
```

Comment:
This fixes some of the sporadic failures on CI, so I'm going to merge even with the style faux pas. /cc @Fryguy